### ISSUE TITLE
Search only within current scope

### ISSUE THREAD

**Issue by hqkhan** [echasnovski/mini.nvim#158]
Hello! I wanted to ask if there's already functionality to only search within scope using mini.indentscope's API. I was hoping to pass current scope's lines into [fzf-lua](https://github.com/ibhagwan/fzf-lua) and search only within that scope. I couldn't quite figure it out. I did fine `require('mini.indentscope').get_scope()` function that I can use but it did not return what I thought it would. Thanks!

**Comment by echasnovski** [comment]
No, unfortunately, there is no functionality for searching only within scope. You can use other tools for that and 'ibhagwan/fzf-lua` is a fine choice.

I don't really know how to do that though. If I understand you correctly, it would be enough for you to get actual lines within current indent scope? If so, here is a function which can do that:

```lua
_G.get_scope_lines = function()
  local scope = MiniIndentscope.get_scope()
  if scope.border.indent < 0 then return {} end
  return vim.api.nvim_buf_get_lines(scope.buf_id, scope.border.top - 1, scope.border.bottom, true)
end
```

It returns lines inside scope including its border. If you want to exclude borders, replace `scope.border` with `scope.body`. Note that it will differ for lines with zero indent: with border it will return to lines, without it - all buffer lines. Tweak this for your liking.

Does this answer your question?

**Comment by hqkhan** [comment]
That does work! It returns me all the lines in the current scope as expected. Now, I just have to pass it to fzf-lua properly to get it to work. Thank you so much!

I had one other question about scopes within comments. Is there a way to get scopes working for comments as well? Sometimes, I'll comment large body of code and uncommenting them would be easy. I can use relative numbers for that for sure but thought I'd ask.
